<html>

<head>
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<title>Genome browser help</title>
<meta name="generator" content="Namo WebEditor">
</head>

<body bgcolor="white" text="black" link="blue" vlink="purple" alink="red">
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><b><font face="Arial"><span style="font-size:14pt;">Overall</span></font></b><font face="Arial"><span style="font-size:10pt;"></span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">STAR browser system is an integrated platform for visualization&nbsp;of next generation sequence(NGS) data. The system provides&nbsp;ability to handle and visualize&nbsp;huge amounts of&nbsp;NGS data. </span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Get started</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Register account-&gt;log into system</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">After log into the system, user&nbsp;can browse all public data tracks (accessible by&nbsp;all)&nbsp;and private tracks (only accessible by&nbsp;owner). User can also log into system by&nbsp;&quot;guest login&quot; option, which means only public data tracks are accessible, and the configurations could disappear&nbsp;due to refreshed web browser cookies.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Browse tracks</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">All available tracks are shown in upper right table. Using below&nbsp;options to search desired tracks:</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&quot;All tracks&quot; button: all available tracks that can be accessed by this user.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&quot;SDEC tracks&quot; button: specific data sets for UCSD Epigenomic Roadmap project.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&quot;My Tracks&quot; button: all tracks created or owned by this user.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Search by keywords: perform fulltext searching by input keywords(and/or logical operations can be used).</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&quot;Tracks&quot; panel in left tab panel: search matched tracks by combined options such as cell, center, experiment type, histone marks and organism.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Sort tracks</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Click table&nbsp;title to sort tracks by this column.</span></font></P>
<DIV>
    <p><img src="manual/sort.jpg" width="633" height="61" border="0"></p>
</DIV>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Remove selected tracks</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Mark checkbox associated with selected tracks-&gt; click &quot;Remove selected&quot; at&nbsp;the bottom of the table region</span></font></P>
<DIV>
    <p><img src="manual/remove.jpg" width="501" height="137" border="0"></p>
</DIV>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Edit track information</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Only tracks owned by this user can be modified. Track information will be displayed in &quot;Editor&quot; tab panel in right bottom region.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Search tracks-&gt;change tabpanel to &quot;Editor&quot;-&gt;click &quot;Add to track pool/Editor&quot;/Drag to editor&quot;-&gt;Modify-&gt; click &quot;Save Track&quot;:</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><img src="manual/img1.gif" border="0"></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Browse configurations.</span><span style="font-size:10pt;">&nbsp;All available configurations are shown in &quot;Configs&quot; tabpanel. </span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><img src="manual/img2.gif" border="0"></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Search configuration:</span><span style="font-size:10pt;"> input keywords &nbsp;in &quot;search input box&quot; and enter.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Sort configuration:</span><span style="font-size:10pt;"> click column head titles</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Remove configuration:</span><span style="font-size:10pt;"> marks checkbox associated with &nbsp;configurations, and then click &quot;Remove selected&quot;.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">View configuration:</span><span style="font-size:10pt;"> click &quot;View&quot; button associated with the configuration. All the tracks in the configuration will be visualized.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Edit configuration:</span><span style="font-size:10pt;"> click &quot;Edit&quot; button associated with the configuration. Track information will be loaded in &quot;Track pool&quot; table</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Create/Update configuration</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><img src="manual/img3.gif" border="0"></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Create Configuration:</b> &nbsp;Browse tracks-&gt;Mark selected tracks-&gt;Select track pool-&gt;Drag to Track pool/click &quot;Add to track pool/Editor&quot;-&gt;Drag to reorder tracks/Remove selected tracks-&gt;change configuration name/description-&gt;click &quot;Create configuration&quot;</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Update Configuration:</b> &nbsp;Click &quot;Edit&quot; in &quot;Configs&quot; list-&gt;Drag to reorder tracks/Add new tracks/Remove selected tracks-&gt;change configuration name/description-&gt;click &quot;Update&quot; button to overwite and update this configuration</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Upload new data tracks</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Track Meta information:</b> &nbsp;</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><img src="manual/img4.gif" border="0"></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Access permission:</b> </span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Private: tracks can only be seen by &nbsp;this user</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Group: If user specifies a group account during registration, tracks will be accessible to all users belong to the group account</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Public: tracks will be released to all users</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Track type:</b></span></font><b></b></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Short reads: This type is usually used to visualize aligned ChIP-Seq/RNA-Seq data, such as Eland, Bowtie, BED files, etc. The signals will be displayed as histogram style at higher zooming ratio, while nucletides will be displayed at lower zooming ratio.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">CG/CHG/CHH: It&nbsp;is used to display methylation sites called from Methyl-Seq/MeDIP-Seq/RRBS/MRE-Seq experiments. A typical result file usually like this:</span></font></P>
<p style="line-height:1; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">chr1 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;469 &nbsp;&nbsp;&nbsp;&nbsp;- &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CG &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;14 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;15</span></font></p>
<p style="line-height:1; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">chr1 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;471 &nbsp;&nbsp;&nbsp;&nbsp;+ &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CHG &nbsp;&nbsp;&nbsp;15 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;15</span></font></p>
<p style="line-height:1; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">chr1 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;484 &nbsp;&nbsp;&nbsp;&nbsp;- &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CHH &nbsp;&nbsp;&nbsp;16 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;18</span></font></p>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Intensity: &nbsp;It's&nbsp;used to visualize sequence data and real&nbsp;values.The signals will be displayed as histogram style at all&nbsp;zooming levels.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Organism:</b></span></font><b></b></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">This is related to proper gene model information, such as chromosomes, genome length, etc.&nbsp;</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Institute:</b></span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Input organization information.&nbsp;It's helpful when searching &nbsp;desired tracks.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>Data source:</b></span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Data can be directly upload to STAR </span></font><font face="Arial"><span style="font-size:10pt;">server, or distributed in users local machine(remote server). If data is processed and stored in local machine (see server setup section), only data fetchers are requried to provide data track information:</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><img src="manual/img5.gif" border="0"></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Examples for track name:</span></font></P>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><span style="font-size:10pt;"><font face="Arial">bing_T2_eln.heli</font></span></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><span style="font-size:10pt;"><font face="Arial">bing_T2_input.heli</font></span></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><span style="font-size:10pt;"><font face="Arial">ryancms_T2_eln.heli</font></span></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><span style="font-size:10pt;"><font face="Arial">ryancms_T2_input.heli</font></span></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Examples for track URL:</span></font><font face="Arial"><span style="font-size:10pt;"></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">http://tabit.ucsd.edu/fetchers/xue/bing_T2_eln.heli.php</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">http://tabit.ucsd.edu/fetchers/xue/bing_T2_input.heli.php</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">http://tabit.ucsd.edu/fetchers/xue/ryancms_T2_eln.heli.php</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">http://tabit.ucsd.edu/fetchers/xue/ryancms_T2_input.heli.php</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Upload data files:</span></font><font face="Arial"><span style="font-size:10pt;"></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><img src="manual/img6.gif" border="0"></p>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Prepare data files and provide format information for&nbsp;direct upload. </span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">This module requires <b>Adobe flash plugin</b>. If you can not see 'upload' button, please check whether your web browser blocks flash plugin.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Supported file format: <b>BED, WIG, Eland, Bowtie, Mapview, SAM, GFF, self defined</b></span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">If self defined is selected, users need to input&nbsp;a string to specify&nbsp;which columns should be extracted from tab-delimited text file. The string looks like: chromosome index :start index :end index :strand index : value index : sequence index, 
and each index number is separated by ':'. For example: 1:2:3:4:6:-1
 means chromosome, start position, end position, strand and value will 
be extracted from column 1, 2, 3, 4 and 6, respectively, and there's no 
sequence information.</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Compressed files supported: <b>.TAR, .GZ, .TAR.GZ, .ZIP</b></span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">File format is also examined on server side. If file does not pass the format check, it will be ignored. </span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>After data files are uploaded to server, or data fether information is provided, click 'Submit' to complete. Users should see those tracks in the track list.</b></span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;">&nbsp;</P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Setup remote&nbsp;data</span></font></P>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">To help building up local server and data processing tools. </span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Please download &nbsp;<a href="manual/star-browser.tar.gz">star-browser.tar.gz</a> and follow instructions:</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">--------------------------------------------------------------------------------&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Requirements:</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">* GNU C++</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">* Berkeley DB</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">--------------------------------------------------------------------------------</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>1. Install Berkeley DB(Oracle)</b></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(1) tar xzvf db-4.6.18.tar.gz</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(2) cd &nbsp;db-4.6.18/build_unix/ (Select appropriate directory based on OS type)</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(3) ../dist/configure --prefix=/usr/local/berkeleydb/ --enable-cxx &nbsp;&nbsp;(BDB install directory)</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(4) make</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(5) make install</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(6) copy shared libs to system directories</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">copy /usr/local/berkeleydb/lib/*.so /usr/lib/</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">copy /usr/local/berkeleydb/lib/*.so /usr/lib64/</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">--------------------------------------------------------------------------------</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>2. Install STAR package</b></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Suppose all above packages have been installed and been configured successfully,</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">The installation instructions of this package will be:</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">cd src &amp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">vim Makefile</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">CXXFLAGS += -I/usr/local/berkeleydb/include/</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">LDFLAGS := -L/usr/local/berkeleydb/lib/ -ldb_cxx -lpthread</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Modify above two lines if you have different BDB installation directory.</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">make &amp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">make install &amp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">You will see three program:</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">data processing: bin/ngsproc,</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">mini web server: www/websrv,</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">CGI program for data fetcher: www/cgi-bin/fetcher.cgi</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">--------------------------------------------------------------------------------</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>3. Start web server</b></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(1) Modify www/cgi-bin/fetcher.conf</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;&nbsp;&nbsp;&nbsp;List all directories which contain you processed data by ngsproc</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(2) ./websrv &amp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;&nbsp;&nbsp;&nbsp;Mini web server will be started and monitor on port 8080 (<b>do not let firewall block this port</b>)</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">--------------------------------------------------------------------------------</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>4. Process mapped NGS data</b></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Usage:./bin/ngsproc [options ....]</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-h --help Display this usage information</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-a --action &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;database operation, in [create|query|drop|level|count]</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-d --db-dir &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Database directory</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-t --table &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Database table you want to operate</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-i --input &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Input data filename, only for 'create'</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-f --format &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Data format, in [BED|WIG|BOWTIE|BOWTIESAM|SAM|ELAND|ELANDMULTI|ELANDEXPORT|GFF|MAPVIEW]</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-s --start &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Start position, only for 'query', default 0</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-e --end &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;End position, only for 'query', default 100000</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">-c --chrom &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Chromsome index, default 1</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Examples:</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(1) ./bin/ngsproc -a create -t H3K4me1_tbl -f bed -d /data/ngs-data -i /usr/tom/H3K4me1.bed</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Create BDB table from input H3K4me1.bed file. Table name: H3K4me1_tbl; Table path: /data/ngs-data</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">File format: BED</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(2) ./bin/ngsproc -a create -t h1.H3K4me3 -f 1:2:3:-1:5:-1 -d /data/ngs-data -i /usr/tom/H3K4me3.txt</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Create BDB table from input H3K4me3.txt file. Table name: h1.H3K4me3; Table path: /data/ngs-data</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">File format: 1:2:3:-1:5:-1, means(chromsome, start, end, strand, value, sequence) will be extracted</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">from those columns(input file contains many TAB delimited columns). In this example, chromosome from</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">column 1, start postion from column 2, end position from column 3, value from column 5, and there's no</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">strand information(regard as positive strand) and sequence information.</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(3) ./bin/ngsproc -a query -t H3K4me1_tbl -d /data/ngs-data -s 0 -e 100000 -c 2</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Query previously processed reads on chromosome 2, and the region is between 0 and 100000</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">This can be used to check if your data are properly processed.</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">(4) ./bin/ngsproc -a level -d /data/ngs-data</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">This will levelize all your data tables(tracks) under this directory. It's very important to speed up</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">data retrieving, especially when browsing at high zooming level.</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">--------------------------------------------------------------------------------</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"><b>5. Submit your data fetchers</b></span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Log into http://wanglab.ucsd.edu/star/browser/ and use &quot;upload&quot; module to submit your data tracks</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Get track names from track.title, and you can modify them</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">Get track URLs from track.url, Modify &quot;localhost&quot; to your server domain</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">http://localhost:8080/cgi-bin/fetcher.cgi?table=dbname</span></font></p>
<p style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;">&nbsp;</span></font></p>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:14pt;">Visualization</span></font></P>
<P style="line-height:100%; margin-top:0; margin-bottom:0;"><font face="Arial"><span style="font-size:10pt;"> click &quot;View&quot; button to explore your tracks.</span></font></P>
</body>

</html>
